src/metadataGeneration/typeResolver.ts

```diff
@@ -207,6 +207,11 @@ export class TypeResolver {
       }
     }
 
+    if (ts.isTypeOperatorNode(this.typeNode)) {
+      const type = this.current.typeChecker.getTypeFromTypeNode(this.typeNode);
+      return new TypeResolver(this.current.typeChecker.typeToTypeNode(type)!, this.current, this.typeNode, this.context, this.referencer).resolve();
```

Comment:
this will most likely fail for more complex scenarios, need to try/catch

Reply:
okay, and what do we want to do if it fails?

Reply:
some error message that highlights the node that causes the issue:
```ts
 throw new GenerateMetadataError(
              `Could not determine the keys of type ${this.current.typeChecker.typeToString(this.typeNode.type)} `,
              this.typeNode,
            );
```